swap/peer.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethersphere/swarm/p2p/protocols"
+)
+
+var ErrSigDoesNotMatch = errors.New("signature does not match")
+var ErrDontOwe = errors.New("no negative balance")
+
+// Peer is a devp2p peer for the Swap protocol
+type Peer struct {
+	*protocols.Peer
+	swap *Swap
+}
+
+func NewPeer(p *protocols.Peer, s *Swap) *Peer {
+	return &Peer{
+		Peer: p,
+		swap: s,
+	}
+}
+
+// handleMsg is for handling messages when receiving messages
+func (sp *Peer) handleMsg(ctx context.Context, msg interface{}) error {
+	switch msg := msg.(type) {
+
+	case *ChequeRequestMsg:
+		return sp.handleChequeRequestMsg(ctx, msg)
```

Comment:
Seems like you are type asserting on `msg`, so you can have the `handleChequeRequestMsg` take a concrete type (`*ChequeRequestMsg`), rather than `interface{}`, and you don't have to check for that in the handler.

Reply:
The handlers have been updated to have concrete `msg` types as params.